spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/SpannerPersistentEntityImpl.java

```diff
@@ -379,9 +385,22 @@ public PersistentPropertyAccessor getPropertyAccessor(Object object) {
 			public void setProperty(PersistentProperty property,
 					@Nullable Object value) {
 				if (property.isIdProperty()) {
-					throw new SpannerDataException(
-							"Setting the primary key directly via the Key ID property is not supported. "
-									+ "Please set the underlying column properties.");
+					SpannerPersistentEntity owner = (SpannerPersistentEntity) property.getOwner();
+					SpannerPersistentProperty[] primaryKeyProperties = owner.getPrimaryKeyProperties();
+
+					Key keyValue = (Key) value;
+					ArrayList<Object> keyParts = new ArrayList<>();
+					keyValue.getParts().forEach(keyParts::add);
+					if (keyParts.size() != primaryKeyProperties.length) {
```

Comment:
Is it possible to check property names, too, not just the number?

Reply:
I don't think we can. Key parts don't have names.